src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java

```diff
@@ -0,0 +1,157 @@
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Envelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ProcessEvent;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ProcessEventRepository;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.ScannableItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.Status;
+import uk.gov.hmcts.reform.bulkscanprocessor.exceptions.EnvelopeNotFoundException;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.common.DocumentType;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.common.Event;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.ocr.OcrData;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.ocr.OcrDataField;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static uk.gov.hmcts.reform.bulkscanprocessor.helper.EnvelopeCreator.envelope;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnvelopeFinaliserServiceTest {
+
+    @Mock
+    private EnvelopeRepository envelopeRepository;
+
+    @Mock
+    private ProcessEventRepository processEventRepository;
+
+    private EnvelopeFinaliserService envelopeFinaliserService;
+
+    @Before
+    public void setUp() {
+        envelopeFinaliserService = new EnvelopeFinaliserService(
+            envelopeRepository,
+            processEventRepository
+        );
+    }
+
+    @Test
+    public void finaliseEnvelope_should_update_envelope_status_clear_all_ocr_data() {
+        // given
+        UUID envelopeId = UUID.randomUUID();
+
+        Envelope envelope = envelope(
+            "JURISDICTION1",
+            Status.NOTIFICATION_SENT,
+            createScannableItemsWithOcrData(3)
+        );
+
+        given(envelopeRepository.findById(envelopeId)).willReturn(Optional.of(envelope));
+
+        // when
+        envelopeFinaliserService.finaliseEnvelope(envelopeId);
+
+        // then
+        verify(envelopeRepository).findById(envelopeId);
+        ArgumentCaptor<Envelope> envelopeCaptor = ArgumentCaptor.forClass(Envelope.class);
+        verify(envelopeRepository).save(envelopeCaptor.capture());
+        verifyNoMoreInteractions(envelopeRepository);
+
+        Envelope savedEnvelope = envelopeCaptor.getValue();
+        assertThat(savedEnvelope.getStatus()).isEqualTo(Status.COMPLETED);
```

Comment:
that's cool, but maybe it would be also worth having an integration test (ie read from db after this service does its thing)

Reply:
There will be a functional test for this functionality.

Reply:
... although it's nothing but db changes. Will make these integration tests.

Reply:
done